synapse/handlers/directory.py

```diff
@@ -100,6 +101,16 @@ def create_association(self, requester, room_alias, room_id, servers=None,
 
         user_id = requester.user.to_string()
 
+        rooms_for_user = yield self.store.get_rooms_for_user(user_id)
+
+        if len(room_alias.to_string()) > 255:
+            raise SynapseError(
```

Comment:
probably wants to be a 400?

`raise SynapseError(400, "words", Codes.INVALID_PARAM)`